source/interact-data/transaction.txt

```diff
@@ -0,0 +1,283 @@
+
+#. Starts a transaction on the session.
+
+#. Performs the specified data changes.
+
+#. Commits the transaction if no errors occur or ends the transaction if
+   there is an error. See the :ref:`` section to learn more an
+
+#. Closes the session.
+
+If your transaction is committed, {+odm+} calls any ``after_commit``
+callbacks for all objects modified inside the transaction. If there is
+an error and the transaction is rolled back, {+odm+} calls any
+``after_rollback`` callbacks for all objects modified inside the
+transaction.
+
+Example
+~~~~~~~
+
+This example uses the following models to represent documents that
+describe books and films:
+
+.. literalinclude:: /includes/interact-data/transaction.rb
+   :start-after: start-example-models
+   :end-before: end-example-models
+   :language: ruby
+   :dedent:
+
+The following code demonstrates how to perform a transaction on
+different objects to change data in multiple collections:
+
+.. literalinclude:: /includes/interact-data/transaction.rb
+   :start-after: start-txn-operations
+   :end-before: end-txn-operations
+   :language: ruby
+   :dedent:
+
+Client Limitations
+~~~~~~~~~~~~~~~~~~
+
+Only operations on the same client are in scope of a transaction,
+because each transaction is attached to a specific client. Ensure
+that you use objects from same client inside the transaction method
+block.
+
+The following example defines model classes that use different clients
+and demonstrates how operations are run based on the origin client:
+
+.. literalinclude:: /includes/interact-data/transaction.rb
+   :start-after: start-different-clients
+   :end-before: end-different-clients
+   :language: ruby
+   :dedent:
+
+.. note::
+   
+   When you call the ``transaction()`` method on the ``{+odm+}`` module,
+   {+odm+} creates the transaction by using the ``:default`` client.
+
+Ending Transactions
+~~~~~~~~~~~~~~~~~~~
+
+Any exception raised inside the transaction method block ends the
+transaction and rolls back changes. Usually, {+odm+} displays the
+exception, except for the ``Mongoid::Errors::Rollback`` exception. Implement
+this exception in your application to explicitly end the transaction
+without passing on an exception.
+
+Callbacks
+~~~~~~~~~
+
+This transaction API introduces the ``after_commit`` and
+``after_rollback`` callbacks.
+
+{+odm+} triggers the ``after_commit`` callback for an object that was
+created, saved, or deleted in the following cases:
+
+- After the transaction is committed if the object was modified inside
+  the transaction.
+
+- After the object is persisted if the object was modified before
+  the transaction block.
+
+The ``after_commit`` callback is triggered only after all
+other callbacks are executed successfully. Therefore, if an object is
+modified without a transaction, it is possible that the object was
+persisted, but the ``after_commit`` callback was not triggered, for
+example, if {+odm+} raised an exception in the ``after_save`` callback.
```

Comment:
S to split up sentences and present tense. Also, could you provide a reference for ``after_save``?
```suggestion
other callbacks are executed successfully. Therefore, if an object is
modified without a transaction, it is possible that the object is
persisted, but the ``after_commit`` callback is not triggered. For
example, if {+odm+} raises an exception in the ``after_save`` callback, 
the ``after_commit`` callback will not be triggered because [blablabla].
```